src/status_im/contexts/wallet/common/utils.cljs

```diff
@@ -314,3 +319,40 @@
                      priority   (get token-priority (:token token) 999)]
                  [(- fiat-value) priority]))
              calculated-tokens)))
+
+(defn on-asset-or-address-confirm
+  [{:keys [address address-prefix recipient collectible]}]
+  (let [[prefix]                    (when address (split-prefix-and-address address))
+        receiver-networks           (network-utils/resolve-receiver-networks
+                                     {:prefix           (or address-prefix prefix)
+                                      :testnet-enabled? (rf/sub [:profile/test-networks-enabled?])
+                                      :goerli-enabled?  (rf/sub [:profile/is-goerli-enabled?])})
+        collectible-tx?             (or (some? collectible)
+                                        (send-utils/tx-type-collectible?
+                                         (rf/sub [:wallet/wallet-send-tx-type])))
+        send-transaction-data       (rf/sub [:wallet/wallet-send])
```

Comment:
It's not common to subscribe in event callbacks :thinking:  I think we should avoid it. Please check https://day8.github.io/re-frame/FAQs/UseASubscriptionInAnEventHandler/#the-root-problem

Reply:
Thanks @ulisesmac! Refactored the component to an event